# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/zstd/i486-no-cpuid.patch
# Copyright (C) 2021 - 2024 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

--- zstd-1.5.6/lib/common/cpu.h.orig	2024-03-27 15:40:19.864687115 +0100
+++ zstd-1.5.6/lib/common/cpu.h	2024-03-27 16:10:07.642597063 +0100
@@ -34,6 +34,7 @@
     U32 f1d = 0;
     U32 f7b = 0;
     U32 f7c = 0;
+#ifdef __i486__
 #if defined(_MSC_VER) && (defined(_M_X64) || defined(_M_IX86))
 #if !defined(__clang__)
     int reg[4];
@@ -133,6 +134,7 @@
               : "edx");
     }
 #endif
+#endif
     {
         ZSTD_cpuid_t cpuid;
         cpuid.f1c = f1c;
